app/src/main/java/edu/wpi/first/shuffleboard/app/dnd/ResizeUtils.java

```diff
@@ -0,0 +1,37 @@
+package edu.wpi.first.shuffleboard.app.dnd;
+
+import edu.wpi.first.shuffleboard.app.components.Tile;
+
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Predicate;
+
+import javafx.scene.Node;
+
+/**
+ * A utility class for dealing with resizing tiles.
+ */
+public final class ResizeUtils {
+
+  public static final Predicate<Node> isResizedTile = n -> n instanceof Tile && isResizedTile((Tile) n);
+
+  private static final AtomicReference<Tile<?>> currentTile = new AtomicReference<>(null);
```

Comment:
This seems a bit concerning. Why is this necessary?

Reply:
It's so that the widget pane can ignore the tile being resized when previewing its layout. Otherwise the highlight would always be red/"invalid" instead of only when it intersects with another tile

Reply:
Not a fan of this but I don't have a better solution that doesn't involve some sort of DI framework.

Eg. The GRIP Drag service: https://github.com/WPIRoboticsProjects/GRIP/blob/1c1a5f6faa310dfd82b576632c46f2e9e00dc728/ui/src/main/java/edu/wpi/grip/ui/dragging/DragService.java